sources/packages/api/src/services/institution/institution.service.ts

```diff
@@ -347,16 +353,75 @@ export class InstitutionService extends RecordDataModelService<Institution> {
     };
   }
 
-  async allUsers(institutionId: number): Promise<InstitutionUser[]> {
-    return this.institutionUserRepo
+  /**
+   * service method to get all institution users with the
+   * given institutionId r.
+   * @queryParm page, page number if nothing is passed then
+   * DEFAULT_PAGE_NUMBER is taken
+   * @queryParm pageLimit, limit of the page if nothing is
+   * passed then DEFAULT_PAGE_LIMIT is taken
+   * @queryParm searchName, user's name keyword to be searched
+   * @queryParm sortField, field to be sorted
+   * @queryParm sortOrder, oder to be sorted
+   * @param institutionId institution id
+   * @returns All the institution users for the given institution
+   * with total count.
+   */
+  async allUsers(
+    searchName: string,
+    sortField: UserFields,
+    sortOrder: FieldSortOrder,
+    institutionId: number,
+    page = DEFAULT_PAGE_NUMBER,
+    pageLimit = DEFAULT_PAGE_LIMIT,
+  ): Promise<[InstitutionUser[], number]> {
+    const institutionUsers = this.institutionUserRepo
       .createQueryBuilder("institutionUser")
-      .leftJoinAndSelect("institutionUser.user", "user")
-      .leftJoin("institutionUser.institution", "institution")
-      .leftJoinAndSelect("institutionUser.authorizations", "authorizations")
-      .leftJoinAndSelect("authorizations.location", "location")
-      .leftJoinAndSelect("authorizations.authType", "authType")
-      .where("institution.id = :institutionId", { institutionId })
-      .getMany();
+      .select([
+        "institutionUser.id",
+        "user.email",
+        "user.firstName",
+        "user.lastName",
+        "user.userName",
+        "user.isActive",
+        "authorizations",
+        "authorizations.authType",
+        "authorizations.id",
+        "authType.role",
+        "authType.type",
+        "authorizations.location",
+        "location.name",
+      ])
+      .innerJoin("institutionUser.user", "user")
+      .innerJoin("institutionUser.institution", "institution")
+      .leftJoin("institutionUser.authorizations", "authorizations")
+      .leftJoin("authorizations.location", "location")
+      .leftJoin("authorizations.authType", "authType")
+      .where("institution.id = :institutionId", { institutionId });
+
+    // search by user's name
+    if (searchName) {
+      institutionUsers.andWhere(
+        "user.firstName Ilike :searchUser OR user.lastName Ilike :searchUser",
+        {
+          searchUser: `%${searchName}%`,
+        },
+      );
+    }
+    // sorting
+    if (sortField === UserFields.DisplayName && sortOrder) {
```

Comment:
is it possible to add inner brackets here for better readability. ?

Reply:
instead of using multiple if, why can we use map concept like, 

public interface UserFields {
  firstName:  "first.name",
  lastName: "last.name",
 email: "user.email",
}

and use like 

if(sortField){
institutionUsers.orderBy(UserFields[sortField] , sortOrder); //have default sort order and use it only when sortField has value
}

Reply:
mm, here `sortField` and different than the DB table name. `sortField`  is the name in the data table. so, we can't directly map. If you want, we can have a quick sync up